skrub/_reporting/_table_report.py

```diff
@@ -42,6 +41,11 @@ def _check_max_cols(max_plot_columns, max_association_columns):
 
     return max_plot_columns, max_association_columns
 
+def lazy_dataframe_exception(dataframe):
+    if sbd.is_polars(dataframe) and sbd.is_lazyframe(dataframe):
+        raise ValueError(
+            "Cannot automatically collect a lazyframe, this might lead to issues!"
+        )
```

Comment:
Maybe we can have a more explicit error message:
```suggestion
        raise ValueError(
            "The TableReport does not support lazy dataframes. Please call `.collect()` to use the TableReport on the current dataframe."
        )
```

(note that if you accept this suggestion it'll break the pre-commit checks, so you'll have to break the line I wrote so it's on multiple lines)

Also a clarification: there is nothing that prevents us from collecting the dataframe ourselves, but as we discussed, this could lead to problems so we intentionally avoid collecting the lazyframe.